File: device-inventory-ui/src/app/device.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class DeviceService {
  private apiUrl = 'http://localhost:8080/api/device'; // Your backend URL

  constructor(private http: HttpClient) {}

  // Save a device
  saveDevice(device: any): Observable<any> {
    return this.http.post(`${this.apiUrl}/save`, device);
  }

  // Get a device by ID
  getDevice(deviceId: number): Observable<any> {
    return this.http.get(`${this.apiUrl}/${deviceId}`);
  }

  // Modify a device
  modifyDevice(id: number, device: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/modify/${id}`, device);
  }

  // Delete a device
  deleteDevice(deviceId: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/delete/${deviceId}`);
  }
}

File: backend/src/main/java/com/example/modified_assignment_backend/service/DeviceService.java
package com.example.modified_assignment_backend.service;

import java.util.Map;

import com.example.modified_assignment_backend.entity.Device;

public interface DeviceService {
    // Method to save a device
    public Map<String, Object> saveDevice(Device device);

    // Method to retrieve a device by its ID
    public Map<String, Object> getDevice(Long deviceId);

    // Method to modify a device by its ID
    public Map<String, Object> modifyDevice(Long id, Device device);

    // Method to delete a device by its ID
    public Long deleteDevice(Long deviceId);
}
